File: api/src/main/java/com/ruby/rubia_server/core/entity/ZApiStatus.java
package com.ruby.rubia_server.core.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ZApiStatus {
    private boolean connected;
    private String session;
    private boolean smartphoneConnected;
    private boolean needsQrCode;
    private String error;
    private Map<String, Object> rawResponse;

    public static ZApiStatus error(String error) {
        return ZApiStatus.builder()
            .connected(false)
            .needsQrCode(true)
            .error(error)
            .build();
    }
}

File: api-dotnet/Rubia.Server/Entities/ZApiStatus.cs
namespace Rubia.Server.Entities;

public class ZApiStatus
{
    public bool Connected { get; set; }
    public string? Session { get; set; }
    public bool SmartphoneConnected { get; set; }
    public bool NeedsQrCode { get; set; }
    public string? Error { get; set; }
    public Dictionary<string, object>? RawResponse { get; set; }

    public static ZApiStatus CreateError(string error)
    {
        return new ZApiStatus
        {
            Connected = false,
            NeedsQrCode = true,
            Error = error
        };
    }
}